src/records.rs

```diff
@@ -500,7 +501,50 @@ impl RecordBatchEncoder {
     }
 }
 
+struct RecordIterator
+{
+    buf: Bytes,
+    batch_decode_info: BatchDecodeInfo,
+    current: i64,
+}
+
+impl Iterator for RecordIterator
+{
+    type Item = Record;
+
+    fn next(&mut self) -> Option<Record>
+    {
+        if self.current >= self.batch_decode_info.record_count as i64
+        {
+            return None;
+        }
+        self.current += 1;
+        Record::decode_new(&mut self.buf, &self.batch_decode_info, 2).ok()
```

Comment:
I think we should return a Result here instead of dropping to an Option.

Reply:
Hi, I still think this is an issue.

I think we need a signature:
```rust
fn next(&mut self) -> Option<Result<Record>>
```

Where:
* `None` indicates the iterator is exhaused.
* `Some(Ok(Record))` indicates a record was successfully parsed.
* `Some(Err(Error))` indicates a record was not succesfully parsed.